test/hexpm_web/controllers/api/user_controller_test.exs

```diff
@@ -210,6 +210,41 @@ defmodule HexpmWeb.API.UserControllerTest do
     end
   end
 
+  describe "GET /api/users/:name/audit_logs" do
+    test "returns the first page of audit_logs for this user" do
+      insert(:audit_log,
+        user: insert(:user, username: "tester"),
+        action: "test.action"
+      )
+
+      conn =
+        build_conn()
+        |> get("/api/users/tester/audit_logs")
+
+      assert [
+               %{
+                 "action" => "test.action"
+               }
+             ] = json_response(conn, :ok)
+    end
+
+    test "returns the second page of audit_logs when params page is 2" do
+      user = insert(:user, username: "tester")
+      insert(:audit_log, user: user, action: "test.second.page")
+      insert_list(10, :audit_log, user: user)
```

Comment:
I have a little performance concern for this test (inserting 11 db records). 
But I don't know a better way to test pagination at controller level. 
Any advice?

(And I'm okay with removing this test)

Reply:
That's very fine. We have way slower tests :)